### PR TITLE
Download icuexportdata in testdata downloader

### LINKED ISSUE #2210: ICU4X data downloader should also download icuexportdata
It appears that the ICU4X data downloader only downloads CLDR data. It should download icuexportdata as well.

### PR DESCRIPTION
**Pull request by sffc**
Fixes #2210

### CODE REVIEW

**Comment by sffc** [comment]
It's not perfect code but my aim was to get something working so I can check it in and move on to what I need it for next.

File: provider/testdata/Cargo.toml [hunk 1]
```diff
@@ -137,7 +137,134 @@ cldr_json_glob = [
 ]
 
 # Git tag or sha1 for the CLDR data used to generate the testdata.
-gitref = "41.0.0"
+cldr_json_gitref = "41.0.0"
+
+icuexportdata_glob = [
```
Comment (sffc) [inline provider/testdata/Cargo.toml:142]: Right now this is just an exhaustive list of everything that was currently being included. We can globify it later if needed.

File: provider/testdata/src/bin/download-sources.rs [hunk 2]
```diff
@@ -47,26 +50,85 @@ impl CldrJsonDownloader<'_> {
...
+}
+
+impl IcuExportDataDownloader<'_> {
+    /// Returns the reqwest client back to the caller for use later
+    async fn download(self, client: &reqwest::Client) -> eyre::Result<IcuExportDataUnzipper> {
+        let url = format!(
+            "https://github.com/{}/releases/download/{}/icuexportdata_{}.zip",
+            self.repo_owner_and_name,
+            self.tag,
+            self.tag.replace('/', "-")
```
Comment (sffc) [inline provider/testdata/src/bin/download-sources.rs:78]: Not super super happy that this is duplicated from datagen but it's fine

File: provider/testdata/src/bin/download-sources.rs [hunk 3]
```diff
@@ -47,26 +50,85 @@ impl CldrJsonDownloader<'_> {
...
+            .by_name(path)
+            .with_context(|| format!("Did not find file in zip: {:?}", &path))?;
+        let local_path = self.root_dir.join(path);
+        fs::create_dir_all(local_path.parent().unwrap())
+            .await
+            .with_context(|| format!("Failed to create dir: {:?}", &local_path))?;
+        let mut file = fs::File::create(&local_path)
+            .await
+            .with_context(|| format!("Failed to create file: {:?}", &local_path))?;
+        // the `zip` crate is not async, so unzip the file into a buffer first
```
Comment (sffc) [inline provider/testdata/src/bin/download-sources.rs:116]: There is an `async_zip` crate but it seems to be early-stage experimental

**Comment by sffc** [comment]
Hmmmm

```
  Uncaught exception in test/data-providers.mjs
  ReferenceError: wasm is not defined
  › file://lib/diplomat-runtime.js:92:3
  › FinalizationRegistry.cleanupSome (<anonymous>)
  ✖ test/data-providers.mjs exited with a non-zero exit code: 1
```

**Comment by sffc** [comment]
Well of course, `wasm` is undefined

```javascript
const DiplomatBuf_finalizer = new FinalizationRegistry(({ ptr, size, align }) => {
  wasm.diplomat_free(ptr, size, align);
});
```

Did we never run that code path before? I don't know why this change would cause that code path to start running

**Comment by sffc** [comment]
It's triggered by the finalization registry. So it's probably random.

@QnnOkabayashi

**Review by robertbastian — APPROVED** [review]
Have you not run it yet or is there actually no data diff?

**Comment by QnnOkabayashi** [comment]
> Well of course, `wasm` is undefined
> 
> ```js
> const DiplomatBuf_finalizer = new FinalizationRegistry(({ ptr, size, align }) => {
>   wasm.diplomat_free(ptr, size, align);
> });
> ```
> 
> Did we never run that code path before? I don't know why this change would cause that code path to start running

I ran into this as well with my webpack stuff as well, the fix is coming in my webpack PR later today.

**Pull request merged**